lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -210,6 +213,38 @@ export abstract class ClientApplication {
         }
     }
 
+    /**
+     * Acquires tokens with password grant by exchanging client applications username and password for credentials
+     *
+     * The latest OAuth 2.0 Security Best Current Practice disallows the password grant entirely.
+     * More details on this recommendation at https://tools.ietf.org/html/draft-ietf-oauth-security-topics-13#section-3.4
+     * Microsoft's documentation and recommendations are at:
+     * https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows#usernamepassword
+     *
+     * @param request - UsenamePasswordRequest
+     */
+    async acquireTokenByUsernamePassword(request: UsernamePasswordRequest): Promise<AuthenticationResult | null> {
+        this.logger.info("acquireTokenByUsernamePassword called", request.correlationId);
+        const validRequest: CommonUsernamePasswordRequest = {
+            ...request,
+            ...this.initializeBaseRequest(request)
+        };
+        const serverTelemetryManager = this.initializeServerTelemetryManager(ApiId.acquireTokenByUsernamePassword, validRequest.correlationId!);
```

Comment:
Looks like we should turn off whatever lint rule is raising the "Forbidden non-null assertion" warning/error.

Reply:
We turned this rule on when we turned on `strictNullChecks` to remind us to fix all the non-null assertions in node. We should fix all these warnings and then set the lint rule to error

Reply:
Makes sense, sounds good.

Reply:
Let's log this in ADO @tnorling. All pending work under strict-null-checks.